File: LV.2/기능 개발/solution.py
import math
def solution(progresses, speeds):
    days = []
    answer = []

    def one(a,b):
        x = 0
        x = math.ceil(a/b)
        return x

    for i in range(len(progresses)):
        days.append(one(100 - progresses[i],speeds[i]))

    count = 1
    for i in range(1, len(days)):
        if days[i-count]>=days[i]:
            count += 1
        else:
            answer.append(count)
            count = 1
    answer.append(count)

    return answer

File: LV.2/기능 개발/Solution.java
import java.util.*;

class Solution {
    public static int totalDay(int progress, int speed){
        return (int) Math.ceil((double) (100-progress)/speed);
    }
    
    public static List<Integer> solution(int[] progresses, int[] speeds) {

    List<Integer> answer = new ArrayList<>();
    Queue<Integer> q = new LinkedList<>();
    
    for (int i=0; i<progresses.length; i++) {
        int day = totalDay(progresses[i], speeds[i]);
        q.add(day);
    }

    while (q.size()>0) {
        int count = 1;
        int left = q.poll();
        
        while(!q.isEmpty() && left >= q.peek()){
            count++;
            q.poll();
        }
        answer.add(count);
    }
    
    return answer;
  }
}
